packages/orchestration/src/orchestration-chat-completion-stream-chunk-response.ts

```diff
@@ -0,0 +1,39 @@
+import type { CompletionPostResponseStreaming } from './client/api/schema/index.js';
+import type { TokenUsage } from './index.js';
+
+/**
+ * Azure OpenAI chat completion stream chunk response.
+ */
+export class OrchestrationChatCompletionStreamChunkResponse {
+  constructor(public readonly data: CompletionPostResponseStreaming) {
+    this.data = data;
+  }
+
+  /**
+   * Usage of tokens in the chunk response.
+   * @returns Token usage.
+   */
+  getTokenUsage(): TokenUsage | undefined {
+    return this.data.orchestration_result?.usage;
+  }
+
+  /**
+   * Reason for stopping the completion stream chunk.
+   * @param choiceIndex - The index of the choice to parse.
+   * @returns The finish reason.
+   */
+  getFinishReason(choiceIndex = 0): string | undefined {
```

Comment:
[q] Do we know if orchestration plans to support multi-choice? I guess our solution is more generic, but just wanna be sure that this will not cause the chunk to look differently and we are still trying to access the chunks correctly.